pyproject.toml

```diff
@@ -32,7 +32,7 @@ documentation = "https://dashboard.scale.com/nucleus/docs/api"
 packages = [{include="nucleus"}]
 
 [tool.poetry.dependencies]
-python = "^3.6.2"
+python = ">=3.7,<3.11"
```

Comment:
I think this is OK 
> Because Python 3.6 extended support on Windows and Linux will end on 23 December 2021, 
as long as we make sure to signal to users that they need to upgrade the Python version.

I'm wondering how pip deals with this. If people are running Python 3.6 will pip just refuse to list newer version and upgrade? We need to add some sort of message hinting people to upgrade to Python3.7 +.

We might actually have to list the depends as ">=3.6" and the dev depends as ">=3.7" and add a error message in`__init__` for 3.6 hinting to upgrade.

Reply:
Actually I resolved this by making scipy/numpy part of dev requirements and having those specific ones require higher python versions :)